Improve missing module error for subdirectories (#7405)

ethan-leba opened the pull request:
Fixes https://github.com/python/mypy/issues/7405

ethan-leba commented:
Thanks, would be happy to take any suggestions on the wording of the message!

TH3CHARLie commented on mypy/modulefinder.py:
```diff
@@ -49,10 +49,18 @@ class ModuleNotFoundReason(Enum):
             note = "See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports"
+        elif self is ModuleNotFoundReason.WRONG_WORKING_DIRECTORY:
+            msg = "Cannot find implementation or library stub for module named '{}'"
+            note = ("You may be running mypy in the wrong directory, "
```
Some brainstorm: `You may be running mypy in a subdirectory, mypy should be run in the package root directory`. @JukkaL may have better suggestions.

ethan-leba replied:
Any suggestions on this message, or other changes that need to be made in order for this PR to be ready for merge @JukkaL?

JukkaL commented on mypy/test/testcmdline.py:
```diff
@@ -26,6 +26,14 @@
+    "testFailedImportCannotBeFoundInParent": "main/subdir",
+    "testImportInParentButNoInit": "main/subdir",
+    "testFailedImportCanBeFoundInParentsParent": "main/subdir1/subdir2",
+}
```
I'd prefer having a way of specifying this in the test case, as this is more fragile (consider what happens if one of the test cases is renamed). For example, this could be specified via `# cwd: main/subdir` in the test case description, similar to how we have `# cmd: ...` and similar.  This would also make it easier to understand what the test cases are testing exactly.

This is not absolutely essential, but it would be nice to have. If you decide not to implement this, please add a comment to each test case that describes the working directory and mentions `custom_working_dir`.

ethan-leba replied:
Added `# cwd: ...` to `testcmdline.py` 👍

JukkaL commented on test-data/unit/cmdline.test:
```diff
@@ -476,6 +476,50 @@ ignore_missing_imports = True
+[file main/__init__.py]
+[out]
+main.py:1: error: Cannot find implementation or library stub for module named 'wrongdir'
+main.py:1: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
```
Command line tests are quite expensive to run. Could you reduce the number of test cases to 2, for example? Maybe you can combine tested features.

ethan-leba commented:
Thanks for the review @JukkaL! I addressed your comments, and used @TH3CHARLie's suggestion for the error message. Let me know if there's any more work that needs to be done on this PR!
